internal/app/backend/synchronizer_client.go

```diff
@@ -58,15 +59,35 @@ func (sc *synchronizerClient) evaluate(ctx context.Context, id string, proposals
 		return nil, err
 	}
 
-	resp, err := sc.synchronizer.EvaluateProposals(ctx, &ipb.EvaluateProposalsRequest{
-		Id:      id,
-		Matches: proposals})
-	if err != nil {
-		return nil, err
+	if len(proposals) == 0 {
+		return proposals, nil
+	}
+
+	stream, err := sc.synchronizer.EvaluateProposals(ctx)
+	for _, proposal := range proposals {
+		if err = stream.Send(&ipb.EvaluateProposalsRequest{Id: id, Match: proposal}); err != nil {
```

Comment:
need to close stream? or just error out?

Reply:
See the comment in the evaluator PR.
> I think the code is correct since it performed the required no-leakage actions.
> https://github.com/grpc/grpc-go/blob/6fb38bdee17208f9e2f933614eb6d9052095be22/stream.go#L129.

Reply:
Ah so just handling the error ensures the resources are not leaked? ok.